03 Writing Algorithms/03 Securities/99 Asset Classes/09 Index Options/01 Requesting Data/01 Universes/12 Examples.php

```diff
@@ -1,9 +1,138 @@
+    calls = sorted(calls, key=lambda x: x.Strike)
+    # Buy 1 0DTE call option contract for the SPX index.
+    self.Buy(calls[0].Symbol, 1)</pre>
+</div>
+
+<h4>Example 2: Rolling expired index options</h4>
+
+<p>In this example, we will create a universe of option contracts for the SPX index that expire in 30 to 90 days. From this universe, we allocate 10% of our cash to buy an ATM Call option that expires in 90 days. We hold this until expiration and then roll over to the next 90-day ATM contract using the <a href="https://www.quantconnect.com/docs/v2/writing-algorithms/universes/key-concepts#06-Security-Changed-Events">on security changed</a> event handler. Since index options are European (they will not be exercised before expiration) and cash-settled (even if held until exercised, they only affect the cash book), this approach saves us the extra work of handling option exercise or assignment, as we would with equity options.</p> 
+
+<div class="section-example-container">
+    <pre class="csharp">private Option _indexOption;
+
+public override void Initialize()
+{
+    // Subscribe to the index option and filter to get only the ones expiring in 30-90 days.
+    _indexOption = AddIndexOption("SPX", "SPXW");
+    _indexOption.SetFilter((u) => u.IncludeWeeklys().CallsOnly().Expiration(30, 90));
+}
+
+public override void OnData(Slice slice)
+{
+    // Get option chain data for the canonical symbol.
+    if (!Portfolio.Invested && 
+        slice.OptionChains.TryGetValue(_indexOption.Symbol, out var chain))
+    {
+        // Obtain the ATM call that expires furthest (90 days).
+        var expiry = chain.Max(x => x.Expiry);
+        var atmCall = chain.Where(x => x.Expiry == expiry)
+            .OrderBy(x => Math.Abs(x.Strike - x.UnderlyingLastPrice))
+            .First();
+        // Allocate 10% Capital.
+        SetHoldings(atmCall.Symbol, 0.1m);
+    }
+}
+
+public override void OnSecuritiesChanged(SecurityChanges changes)
+{
+    foreach (var removed in changes.RemovedSecurities)
+    {
+        // Liquidate the contracts that exit the universe (due to expiry).
+        if (Portfolio[removed.Symbol].Invested)
+        {
+            Liquidate(removed.Symbol);
+        }
+    }
+}</pre>
+    <pre class="python">def initialize(self):
+    # Subscribe to the index option and filter to get only the ones expiring in 30-90 days.
+    self._index_option = self.add_index_option("SPX", "SPXW")
+    self._index_option.set_filter(lambda u: u.include_weeklys().calls_only().expiration(30, 90))
+
+def on_data(self, slice: Slice):
+    # Get option chain data for the canonical symbol.
+    if not self.portfolio.invested:
+        chain = slice.option_chains.get(self._index_option.symbol)
+        if chain:
+            # Obtain the ATM call that expires furthest (90 days).
+            expiry = max(x.expiry for x in chain)
+            atm_call = min(chain, key=lambda x: abs(x.strike - x.underlying_last_price))
```

Comment:
`expiry` not used. Please run C# and Python examples. They should have the same results.